controllers/todo.js

```diff
@@ -5,36 +5,192 @@ const { ToDo, Token } = require("../models");
 // HINT: Create a middleware for above :)
 
 const getAllToDo = async (req, res) => {
+    
+  const user = req.user
+  ToDo.find({$or: [{createdBy:user._id},{collaborators:user._id}]},{title:1,createdBy:1},function(err,foundTodos){
+    if(!err && foundTodos.length > 0){
+
+      let todosOfUser=[]
+      let todosCollaboratedTo=[]
+      foundTodos.forEach(function(todo){
+        if(todo.createdBy==user.id){
+          todosOfUser.push({"id":todo._id,"title":todo.title})
+        }else{
+          todosCollaboratedTo.push({"id":todo._id,"title":todo.title})
```

Comment:
@try-catch-stack 
Try using `spread` operator. It is faster. See [here](https://www.measurethat.net/Benchmarks/Show/4223/0/array-concat-vs-spread-operator-vs-push) for more details.